burn-book/src/building-blocks/record.md

```diff
@@ -0,0 +1,55 @@
+# Record
+
+Records are how states are saved with Burn. Compared to most other frameworks, Burn has its own
+advanced saving mechanism that allows interoperability between backends with minimal possible
+runtime errors. There are multiple reasons why Burn decided to create its own saving formats.
+
+First, Rust has [serde](https://serde.rs/), which is an extremely well-developed serialization and
+deserialization library that also powers the `safetensors` format developed by HuggingFace. If used
+properly, all the validations are done when deserializing, which removes the need to write
+validation code. Since modules in Burn are created with configurations, they can't implement
+serialization and deserialization. That's why the record system was created: allowing you to save
+the state of modules independently of the backend in use extremely fast while still giving you all
+the flexibility possible to include any non-serializable field within your module.
+
+**Why not use safetensors?**
+
+Safetensors uses serde with the JSON file format and only supports serializing and deserializing
+tensors. The record system in Burn gives you the possibility to serialize any type, which is very
+useful for optimizers that save their state, but also for any non-standard, cutting-edge modeling
+needs you may have. Additionally, the record system performs automatic precision conversion by using
+Rust types, making it more reliable with fewer manual manipulations.
```

Comment:
The question that naturally comes to mind while reading this is: But are they "safe" like safetensors?

safetensors have a checksum to ensure the data was not corrupted. We could argue that by using gzip we also have a checksum.